Sources/mpc-core-kit-swift/MpcCoreKitSwift.swift

```diff
@@ -15,129 +15,113 @@ import TorusUtils
+    public func loginWithOAuth(singleLoginParams: SingleLoginParams) async throws -> MpcKeyDetails {
+        let loginResponse = try await customAuth.triggerLogin(args: singleLoginParams)
 
-        let sub = SubVerifierDetails(loginType: .web,
-                                     loginProvider: loginProvider,
-                                     clientId: clientId,
-                                     verifier: verifier,
-                                     redirectURL: redirectURL,
-                                     browserRedirectURL: browserRedirectURL,
-                                     jwtParams: jwtParams
-        )
-        
-        let customAuth = CustomAuth(web3AuthClientId: option.Web3AuthClientId, aggregateVerifierType: .singleLogin, aggregateVerifier: verifier, subVerifierDetails: [sub], network: network.toTorusNetwork(), enableOneKey: true)
+        let result = try await login(keyDetails: loginResponse.torusKey, verifier: loginResponse.singleVerifierResponse.userInfo.verifier, verifierId: loginResponse.singleVerifierResponse.userInfo.verifierId)
+        userInfo = loginResponse.singleVerifierResponse.userInfo
 
-        let userData = try await customAuth.triggerLogin()
-        return try await login(userData: userData)
+        return result
+    }
+
+    public func loginWithOAuth(aggregateLoginParams: AggregateLoginParams) async throws -> MpcKeyDetails {
+        let loginResponse = try await customAuth.triggerAggregateLogin(args: aggregateLoginParams)
+        let result = try await login(keyDetails: loginResponse.torusKey, verifier: loginResponse.torusAggregateVerifierResponse.first!.userInfo.verifier, verifierId: loginResponse.torusAggregateVerifierResponse.first!.userInfo.verifierId)
+
+        userInfo = loginResponse.torusAggregateVerifierResponse.first?.userInfo
+
+        return result
     }
 
-    // mneomonic to share
     public func mnemonicToKey(shareMnemonic: String, format: String) throws -> String {
-        // Assuming ShareSerializationModule.deserializeMnemonic returns Data
         let factorKey = try ShareSerializationModule.deserialize_share(threshold_key: tkey!, share: shareMnemonic, format: format)
         return factorKey
     }
 
-    // share to mneomonic
     public func keyToMnemonic(factorKey: String, format: String) throws -> String {
-        // Assuming ShareSerializationModule.deserializeMnemonic returns Data
         let mnemonic = try ShareSerializationModule.serialize_share(threshold_key: tkey!, share: factorKey, format: format)
         return mnemonic
     }
 
-    public func loginWithJwt(verifier: String, verifierId: String, idToken: String, userInfo: [String: Any] = [:]) async throws -> MpcKeyDetails {
-        let singleFactor = SingleFactorAuth(singleFactorAuthArgs: SingleFactorAuthArgs(web3AuthClientId: option.Web3AuthClientId, network: network))
+    public func loginWithJwt(verifier: String, verifierId: String, idToken: String) async throws -> MpcKeyDetails {
```

Comment:
it doesn't seems to be supporting aggregate login yet, also it would be good if we can make function arg interface same as web.

Reply:
@ieow need to add a function to support aggregate login.